vpr/src/place/timing_place_lookup.cpp

```diff
@@ -1167,16 +1167,16 @@ bool directconnect_exists(int src_rr_node, int sink_rr_node) {
 
     //TODO: This is a constant depth search, but still may be too slow
     for (t_edge_size i_src_edge = 0; i_src_edge < rr_nodes[src_rr_node].num_edges(); ++i_src_edge) {
-        int opin_rr_node = rr_nodes[src_rr_node].edge_sink_node(i_src_edge);
+        int opin_rr_node = size_t(rr_graph.edge_sink_node(RRNodeId(src_rr_node), i_src_edge));
 
         if (rr_graph.node_type(RRNodeId(opin_rr_node)) != OPIN) continue;
 
         for (t_edge_size i_opin_edge = 0; i_opin_edge < rr_nodes[opin_rr_node].num_edges(); ++i_opin_edge) {
-            int ipin_rr_node = rr_nodes[opin_rr_node].edge_sink_node(i_opin_edge);
+            int ipin_rr_node = size_t(rr_graph.edge_sink_node(RRNodeId(opin_rr_node), i_opin_edge));
             if (rr_graph.node_type(RRNodeId(ipin_rr_node)) != IPIN) continue;
 
             for (t_edge_size i_ipin_edge = 0; i_ipin_edge < rr_nodes[ipin_rr_node].num_edges(); ++i_ipin_edge) {
-                if (sink_rr_node == rr_nodes[ipin_rr_node].edge_sink_node(i_ipin_edge)) {
+                if (size_t(sink_rr_node) == size_t(rr_graph.edge_sink_node(RRNodeId(ipin_rr_node), i_ipin_edge))) {
```

Comment:
In the for loop (starting from LINE 1169), we can change the data type of 
- ``opin_rr_node`` from ``int`` to ``RRNodeId``
- ``ipin_rr_node`` from ``int`` to ``RRNodeId``

I see in LINE 1174, the ``num_edges()`` remain using an old API. I believe that you have patched these line in another PR #1917 
It means that we should merge #1917 before this PR.
Let me know if this is fine for you or not.